### PR TITLE
Fix all 1.7.7.x controllers and routes to comply with conventions

### PR DESCRIPTION
**Pull request by tdavidsonas88**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Changed only the BO controllers/routes created in 1.7.7.x and not present in 1.7.6.x (to avoid backward compatibility issues) to follow PS naming conventions
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | code quality - naming conventions
| How to test?  | We have modified how some URLs are configured in the BO but only the code. The behavior should be unchanged. Please see list of modified URLs below. They should work just like before this PR.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17528)

List of changed routes:
admin_attachments_edit => /sell/attachments/{attachmentId}/edit
admin_attachments_view => /sell/attachments
admin_attachments_delete => /sell/attachments/{attachmentId}/delete
admin_attachments_create => /sell/attachments/new
admin_credit_slips_generate_pdf => /sell/orders/credit-slips/{creditSlipId}/pdf
admin_monitorings_index => /sell/catalog/monitoring/ [GET]
admin_monitorings_search => /sell/catalog/monitoring/ [POST]

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/international/tax_rules_groups.yml
Comment (matks) [inline src/PrestaShopBundle/Resources/config/routing/admin/improve/international/tax_rules_groups.yml]: Some of our controllers use a common Search action. Other have a dedicated search action. I dont remember which practice is the latest, which one is the oldest. We need to explore that to know the answer whether or not the linter is right here.

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/orders/credit_slips.yml
Comment (matks) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/orders/credit_slips.yml]: It's because this page has a custom behavior. Linter does not apply as the page is not standard. We need a way to tell the linter about unconventional pages, so it does not ring false alarms.

**Comment by matks** [comment]
@tdavidsonas88 If you remove the comments, I think we can go for QA 😄

**Comment by Progi1984** [comment]
> @tdavidsonas88 If you remove the comments, I think we can go for QA smile

Yeap I fixed Travis.

**Comment by tdavidsonas88** [comment]
> @tdavidsonas88 If you remove the comments, I think we can go for QA

Removed and also added the suggested route column in a table

**Review by Progi1984 — DISMISSED** [review]

**Comment by tdavidsonas88** [comment]
any ideas why Travis integration tests fail?

**Comment by matks** [comment]
```
> @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/Unit/phpunit.xml
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.
...............................................................  63 / 853 (  7%)
............................................................... 126 / 853 ( 14%)
............................................................... 189 / 853 ( 22%)
............................................................... 252 / 853 ( 29%)
............................................................... 315 / 853 ( 36%)
............................................................... 378 / 853 ( 44%)
............................................................... 441 / 853 ( 51%)
............................................................... 504 / 853 ( 59%)
............................................................... 567 / 853 ( 66%)
............................................................... 630 / 853 ( 73%)
............................................................... 693 / 853 ( 81%)
............................................................... 756 / 853 ( 88%)
............................................................... 819 / 853 ( 96%)
..............F

Time: 2.8 seconds, Memory: 38.00 MB

There was 1 failure:
1) Tests\Unit\PrestaShopBundle\Routing\Linter\NamingConventionLinterTest::testLinterThrowsExceptionWhenRouteAndControllerDoesNotFollowNamingConventions with data set #0 ('admin_test_index', Symfony\Component\Routing\Route Object (...))
Failed asserting that exception of type "PrestaShopBundle\Routing\Linter\Exception\NamingConventionException" is thrown.

FAILURES!
Tests: 834, Assertions: 2160, Failures: 1.
Script @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/Unit/phpunit.xml handling the unit-tests event returned with error code 1
```

Probably a test that checks one of our routes and is now failing because we modified the route name. You should be able to reproduce by running `php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/Unit/phpunit.xml` after having created a test database*

*See `php composer.phar create-test-db`

**Comment by tdavidsonas88** [comment]
Thank you @matks for giving me the understanding that it was a unit test failure. It helped me a lot!

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@tdavidsonas88 I approve, waiting for a 2nd approval.

Can you prepare the list of routes and URLs that have been changed to help QA team to test this PR ?

**Comment by tdavidsonas88** [comment]
> @tdavidsonas88 I approve, waiting for a 2nd approval.
> 
> Can you prepare the list of routes and URLs that have been changed to help QA team to test this PR ?

List of changed routes:
admin_attachments_edit => /sell/attachments/{attachmentId}/edit
admin_attachments_view => /sell/attachments
admin_attachments_delete => /sell/attachments/{attachmentId}/delete
admin_attachments_create => /sell/attachments/new
admin_credit_slips_generate_pdf => /sell/orders/credit-slips/{creditSlipId}/pdf
admin_monitorings_index => /sell/catalog/monitoring/ [GET]
admin_monitorings_search => /sell/catalog/monitoring/ [POST]

**Review by PierreRambaud — DISMISSED** [review]

File: src/PrestaShopBundle/Routing/Linter/NamingConventionLinter.php
Comment (jolelievre) [inline src/PrestaShopBundle/Routing/Linter/NamingConventionLinter.php]: I prefer keeping the exception, but you could add a new parameter in this exception to get back the expected route name

**Review by jolelievre — CHANGES_REQUESTED** [review]
The initial exception was a good idea It just needs to be improved a little

File: src/PrestaShopBundle/Routing/Linter/NamingConventionLinter.php
Comment (jolelievre) [reply src/PrestaShopBundle/Routing/Linter/NamingConventionLinter.php]: Sorry by `I prefer` (bad use of words) I meant we want to avoid having methods with multi type returns, the scope of the function is then not instantly clear and requires to check the type returned by the function to check if it succeeded or not
Rather a smart use of exception is much clearer which is why I suggested to stick to it, but add the required information in it Then it can be caught and used for the appropriate purpose :)

**Review by jolelievre — DISMISSED** [review]
Thank you @tdavidsonas88

**Review by matks — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
Hi fellows,

don't want to hurt anyone here, but changing routes names is actually a BC break.

People in modules are allowed to override routes, like everything that is part of the public API.

The fact that you need to update integration tests is proving my point, same apply for https://github.com/PrestaShop/PrestaShop/pull/17482 (but that one is so BC-broken that I've stopped to review it: I'm pretty sure it's a private joke between you xD: you should notice it btw ^^). 

Cheers

**Comment by matks** [comment]
> Hi fellows,
> 
> don't want to hurt anyone here, but changing routes names is actually a BC break.

We are changing the route names of 1.7.7 controllers, not released yet. 1.7.6 controllers do not change and if one day we change them, we'll provide aliases.

**Comment by mickaelandrieu** [comment]
fair enough ! Ignore my comment 👍

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @tdavidsonas88

**Pull request merged**
